### PR TITLE
FIX-#5776: fix IndexError when concatenating dict of series along columns

### LINKED ISSUE #5776: IndexError when concatenating dict of series along columns
Consider following code example:

```python
df = pd.concat(
    {
        s: pd.Series(index=range(3))
        for s in ["a", "b"]
    },
    axis="columns",
)
```

This works just fine on pandas and produces a dataframe with columns `["a", "b"]` and index `[0, 1, 2]` containing all `nan`. Yet the same code fails for modin on Ray with `IndexError: list index out of range` on line `modin/pandas/general.py:458`:

https://github.com/modin-project/modin/blob/8d3db2b4a7fa79716796b33f7cb3673ed729b652/modin/pandas/general.py#L458

I'm running Python 3.9.6, Ray 2.3.0 and Modin 0.19.0 on a Windows machine.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5776 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/general.py [hunk 1]
```diff
@@ -430,6 +430,8 @@ def concat(
         else DataFrame(obj)._query_compiler
         for obj in list_of_objs
     ]
+    if keys is None and isinstance(objs, dict):
```
Comment (anmyachev) [inline modin/pandas/general.py:433]: According to the [documentation](https://pandas.pydata.org/docs/reference/api/pandas.concat.html?highlight=concat#pandas.concat):
```bash
If a mapping is passed, the sorted keys will be used as the keys argument, unless it is passed...
```

Comment (dchigarev) [reply modin/pandas/general.py:433]: so now these lines seem to be unreachable, do we still want to keep them?
https://github.com/modin-project/modin/blob/a8be482e644519f2823668210cec5cf1564deb7e/modin/pandas/general.py#L456-L459

Comment (anmyachev) [reply modin/pandas/general.py:433]: removed

**Review by dchigarev — APPROVED** [review]

**Comment by anmyachev** [comment]
@dchigarev could we merge it?

**Review by YarShev — APPROVED** [review]

**Pull request merged**